compute/vpn_endpoint_v2.go

```diff
@@ -0,0 +1,276 @@
+package compute
+
+import (
+	"fmt"
+)
+
+const (
+	VPNEndpointV2Description   = "vpn endpoint v2"
+	VPNEndpointV2ContainerPath = "/vpnendpoint/v2/"
+	VPNEndpointV2ResourcePath  = "/vpnendpoint/v2"
+)
+
+type VPNEndpointV2sClient struct {
+	ResourceClient
+}
+
+// VPNEndpointV2s() returns an VPNEndpointV2sClient that can be used to access the
+// necessary CRUD functions for VPN Endpoint V2s.
+func (c *Client) VPNEndpointV2s() *VPNEndpointV2sClient {
+	return &VPNEndpointV2sClient{
+		ResourceClient: ResourceClient{
+			Client:              c,
+			ResourceDescription: VPNEndpointV2Description,
+			ContainerPath:       VPNEndpointV2ContainerPath,
+			ResourceRootPath:    VPNEndpointV2ResourcePath,
+		},
+	}
+}
+
+// VPNEndpointV2Info contains the exported fields necessary to hold all the information about an
+// VPN Endpoint V2
+type VPNEndpointV2Info struct {
```

Comment:
The API also returns a `lifecycleState` attribute, which doesn't appear to be in the docs, but we'll need to monitoring the provisioning, update and delete state of the VPN endpoint with a "wait for ready" function, initial VPN endpoint provisioning can take >1 hour.  The values for the `lifecycleState` are: 

- `provisioning` - VPN endpoint is provisioning (initial state on create)
- `ready` - VPN endpoint provisioned
- `updating` - VPN endpoint is being updated (after update request)
- `deleting`  - VPN endpoint is being deleted (after delete request)
- `error` - VPN endpoint provisioning failed (error state)

In addition, the update and delete will operations can be run when the VPN endpoint is in a `ready` or `error` lifecycle state.

Reply:
This has been added